### PR TITLE
feat: added automatic deployment of predeploy when testing

### PR DESCRIPTION
**Pull request by jatZama**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "task:deployOracle": "hardhat task:deployOracle",
     "task:addRelayer": "hardhat task:addRelayer",
     "task:removeRelayer": "hardhat task:removeRelayer",
-    "fhevm:start": "docker run -i -p 8545:8545 --rm --name fhevm ghcr.io/zama-ai/ethermint-dev-node:v0.3.0-5",
+    "fhevm:start": "chmod +x launch-fhevm.sh && ./launch-fhevm.sh",
```
Comment (youben11) [inline package.json]: why run `chmod` on every call? You can push it in github as executable

**Review by immortal-tofu — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (jat9292) [reply package.json:30]: Do you mean if i do `chmod` locally then I push it would work?

Comment (jatZama) [reply package.json]: Do you mean if i do `chmod` locally then I push it would work?

Comment (youben11) [reply package.json]: yes

File: test/asyncDecrypt.ts [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+const ifaceResultCallbackUint32 = new ethers.Interface(['event ResultCallbackUint32' + argEvents2]);
+const ifaceResultCallbackUint64 = new ethers.Interface(['event ResultCallbackUint64' + argEvents2]);
+
+let oracle: OraclePredeploy;
+let firstBlockListening: number;
+
+export const asyncDecrypt = async (): Promise<void> => {
+  firstBlockListening = await ethers.provider.getBlockNumber();
+  // this function will emit logs for every request and fulfilment of a decryption
+  oracle = await ethers.getContractAt('OraclePredeploy', parsedEnv.ORACLE_CONTRAT_PREDEPLOY_ADDRESS);
```
Comment (david-zk) [inline test/asyncDecrypt.ts]: Typo?
`ORACLE_CONTRAT_PREDEPLOY_ADDRESS`
should be
`ORACLE_CONTRACT_PREDEPLOY_ADDRESS`

Comment (jatZama) [reply test/asyncDecrypt.ts]: good catch, this was badly written dozens of times. I pushed a fix.

File: package.json (hunk 1 above)
Comment (jatZama) [reply package.json]: done

**Review by immortal-tofu — APPROVED** [review]

**Pull request merged**